bexpr.go

```diff
@@ -9,12 +9,20 @@ package bexpr
 
 import (
 	"github.com/hashicorp/go-bexpr/grammar"
+	"github.com/mitchellh/pointerstructure"
 )
 
+// HookFn provides a way to translate 1 reflect.Value to another during
+// evaluation by bexpr.  This facilitate making go structures appear in a way
+// that matches the expected JSON Pointers used for evaluation. This is
+// helpful, for example, when working with protocol buffers' well-known types.
+type ValueTransformationHookFn pointerstructure.ValueTransformationHookFn
```

Comment:
I suggest making this a type alias (`type ValueTransformationHookFn = pointerstructure.ValueTransformationHookFn`). That allows users to specify either type (because they're the same) and avoids the cast in evaluate.go. Both can work, but given that this is meant as a passthrough, declaring the type as equivalent instead of a subtype seems like it better matches the intention.